src/internal_rules.jl

```diff
@@ -815,3 +815,17 @@ function EnzymeRules.forward(func::Const{typeof(ldiv!)},
         end
     end
 end
+
+# Ranges
+# Float64 ranges in Julia use bitwise `&` with higher precision
+# to correct for numerical error, thus we put rules over the
+# operations as this is not directly differentiable
+
+getval(x) = hasproperty(x, :val) ? x.val : x
+function EnzymeRules.forward(func::Const{Colon}, ::Type{<:Duplicated}, start::Union{Const, Active}, step::Union{Const, Active}, stop::Union{Const, Active})
```

Comment:
Can you remove the union{const,active} restriction, make Type{<:Duplicated} more general, and in doing so add batchduplicated/duplicatednoneed?

See https://github.com/EnzymeAD/Enzyme.jl/blob/d6a3ddcd4122d45b3b245b436f35b5d81d84fe1c/src/internal_rules.jl#L582 / https://github.com/EnzymeAD/Enzyme.jl/blob/d6a3ddcd4122d45b3b245b436f35b5d81d84fe1c/src/internal_rules.jl#L600 for example